src/components/ListItem.jsx

```diff
@@ -1,5 +1,52 @@
+import { useEffect, useState } from 'react';
+import { updateItem, undoUpdateItem } from '../api/firebase';
+
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listToken, itemId, dateLastPurchased }) {
+	const [wasPurchased, setWasPurchased] = useState(false);
+
+	const handleCheck = () => {
+		if (wasPurchased) {
+			undoUpdateItem(listToken, itemId);
+			setWasPurchased(false);
+		} else {
+			updateItem(listToken, itemId);
+			setWasPurchased(true);
+		}
+	};
```

Comment:
I think there can be a way to toggle the state in one line! Here's a [hint](https://stackoverflow.com/questions/40359800/how-to-toggle-boolean-state-of-a-react-component) :) 

Then if you'd like to apply the comment about `updateItem()`, you'd only need to call it once here and pass that new parameter.